share/wake/lib/system/job.wake

```diff
@@ -318,6 +320,15 @@ export def runJob p = match p
         def _ = wait (\_ badfinish job error) (badlaunch job error)
         job
 
+# Provide a matching function and a replacement resource to override a resource
+# A non matching function should return the original resource
+global topic resourceOverrides: ((label: String) => (origResource: String) => (newResource: String))
+
+def override label defaultResources =
+  def fns = subscribe resourceOverrides
+  def fnsWithLabel resource = map (_ label resource) fns
+  map (\r fnsWithLabel r) defaultResources | flatten
```

Comment:
```suggestion
  def compose f g = f ∘ g  # Really? No ASCII name for the compose function?
  def id x = x
  def composeAll functions = foldl compose id functions

  def fnsWithLabel resource = map (_ label resource) fns
  def overrideResource = composeAll fns
  
  map overrideResource originalResources
```

If I'm reading the code correctly, I think your code will end up duplicating the resource string for any resource that isn't matched by the current override function. For example, if I have two override functions that map `foo/1.0` -> `foo/2.0` and `bar/1.0` -> `bar/2.0`, then if I apply it to a job that has `resources = "foo/1.0", "bar/1.0", Nil`, I'll get `"foo/2.0", "foo/1.0", "bar/1.0", "bar/2.0", Nil` as the final set of resources.

I think what you want to do is iterate over each resource and then apply each override function in a chain.